### ISSUE TITLE
Freeze and crash after clicking Online Queries button

### ISSUE THREAD

**Issue by worachate001** [Stellarium/stellarium#2548]
### Expected Behaviour
Stellarium open the browser to search for information of selected object.

### Actual Behaviour
Program freeze and crash.

No error message in 0.22.* series. In version 1.22.2 this error appears in log file:

`Could not find QtWebEngineProcess`

Note: The plugin works properly in version 0.21.3.

### Steps to reproduce
1. Enable Online Queries plugin
2. Click an object
3. Click Online Queries button

### System
* Stellarium version:  0.22.1, 0.22.2, 1.22.2
* Operating system: Linux Mint 21

**Comment by alex-w** [comment]
Hmm… are you trying an AppImage or something else?

**Comment by worachate001** [comment]
> Hmm… are you trying an AppImage or something else?

Yes, the AppImage for 0.22.* (AppImage for 1.22.2 is not working).

**Comment by alex-w** [comment]
OK, please check an AppImage from here: https://github.com/Stellarium/stellarium/actions/runs/2787633526 (please wait the finish of building)

**Comment by alex-w** [comment]
@worachate001 has your Mint by default X11 session?

**Comment by worachate001** [comment]
> OK, please check an AppImage from here: https://github.com/Stellarium/stellarium/actions/runs/2787633526 (please wait the finish of building)

No different, still freeze and crash when running the AppImage.

**Comment by worachate001** [comment]
> @worachate001 has your Mint by default X11 session?

Yes.

**Comment by gzotti** [comment]
I don't even come as far as proper initialisation (Ubuntu 20.04) :-(  [log.txt](https://github.com/Stellarium/stellarium/files/9250401/log.txt)

**Comment by alex-w** [comment]
This is really annoying me. What about deb packages from our PPA’s?

**Comment by gzotti** [comment]
Uh, I have too many systems. On this particularly it also fails with the same pattern. On the other (U22.04) the ppa version worked a week ago. Did not check the AppImage there. 
I think tomorrow is time for upgrading to U22.04.01 LTS...

**Comment by gzotti** [comment]
OK, something was wrong with my graphics system. After a reboot of U20.04, Stellarium 0.22.2, the AppImage and self-built master all work.

**Comment by alex-w** [comment]
I've removed preparation an AppImage packages now

**Comment by alex-w** [comment]
Our PPA for U22.04 contain Qt5-based version of Stellarium. I can try update debianizer for Qt6-based Stellarium, but... I still use U20.04. 

@gzotti could you share `ldd ./stellarium` from U22.04 for Qt6-based Stellarium?

**Comment by alex-w** [comment]
@gzotti so, the problem in the code or in the packaging?

**Comment by gzotti** [comment]
Bummer. Message is 

Could not find QtWebEngineProcess.

I see several such reports. E.g.
https://stackoverflow.com/questions/67060913/pyinstaller-could-not-find-qtwebengineprocess-exe-for-pyqt5-web-application
https://github.com/pyinstaller/pyinstaller/issues/3890

These describe PyQt issues. Some others indicate a program of this name has now to be installed. I confirm a crash on U22, but will need to investigate further.

**Comment by gzotti** [comment]
Here is the ldd result [ldd-stellarium-qt6.txt](https://github.com/Stellarium/stellarium/files/9253928/ldd-stellarium-qt6.txt)

**Comment by gzotti** [comment]
No crash on Windows.

**Comment by alex-w** [comment]
No crash on macOS too...

**Comment by alex-w** [comment]
Please check Qt6-based Stellarium from our PPA: `ppa:stellarium/stellarium-beta`

**Comment by gzotti** [comment]
OK, a missing package: apt install libqt6webenginecore6-bin

After that:
> qt.webenginecontext: 

GLImplementation: desktop
Surface Type: OpenGL
Surface Profile: CompatibilityProfile
Surface Version: 4.6
Using Default SG Backend: yes
Using Software Dynamic GL: no
Using Angle: no

Init Parameters:
  *  application-name stellarium 
  *  browser-subprocess-path /lib/qt6/libexec/QtWebEngineProcess 
  *  create-default-gl-context  
  *  disable-features DnsOverHttpsUpgrade,ConsolidatedMovementXY,InstalledApp,BackgroundFetch,WebOTP,WebPayments,WebUSB,PictureInPicture 
  *  disable-setuid-sandbox  
  *  disable-speech-api  
  *  enable-features NetworkServiceInProcess,TracingServiceInProcess 
  *  enable-threaded-compositing  
  *  in-process-gpu  
  *  use-gl desktop 

Qt WebEngine resources not found at /share/qt6/resources. Trying parent directory...
Qt WebEngine resources not found at /share/qt6. Trying application directory...
Qt WebEngine resources not found at /usr/bin. Trying fallback directory... The application MAY NOT work.
Installed Qt WebEngine locales directory not found at location /share/qt6/translations/qtwebengine_locales. Trying application directory...
Qt WebEngine locales directory not found at location /usr/bin/qtwebengine_locales. Trying fallback directory... Translations MAY NOT not be correct.
[0804/004948.060073:ERROR:resource_bundle.cc(911)] Failed to load /home/gzotti/.stellarium/qtwebengine_resources_100p.pak
Some features may not be available.
[0804/004948.061343:ERROR:resource_bundle.cc(911)] Failed to load /home/gzotti/.stellarium/qtwebengine_resources_200p.pak
Some features may not be available.
[0804/004948.061404:ERROR:resource_bundle.cc(911)] Failed to load /home/gzotti/.stellarium/qtwebengine_resources.pak
Some features may not be available.
[0804/004948.062289:WARNING:resource_bundle_qt.cpp(119)] locale_file_path.empty() for locale 
[53574:53574:0804/004948.318049:ERROR:extension_system_qt.cpp(121)] Failed to parse extension manifest.
Speicherzugriffsfehler (Speicherabzug geschrieben)

**Comment by gzotti** [comment]
Update from Windows: I ran a debug build. Its logfile also has more info:

qt.webenginecontext: 

GL Type: desktop
Surface Type: OpenGL
Surface Profile: CompatibilityProfile
Surface Version: 4.6
QSG RHI Backend: OpenGL
Using Supported QSG Backend: yes
Using Software Dynamic GL: no
Using Multithreaded OpenGL: yes

Init Parameters:
  *  application-name stellarium 
  *  browser-subprocess-path D:\Qt\6.3.1\msvc2019_64\bin\QtWebEngineProcessd.exe 
  *  create-default-gl-context  
  *  disable-es3-gl-context  
  *  disable-features ConsolidatedMovementXY,InstalledApp,BackgroundFetch,WebOTP,WebPayments,WebUSB,PictureInPicture 
  *  disable-speech-api  
  *  enable-features NetworkServiceInProcess,TracingServiceInProcess 
  *  enable-threaded-compositing  
  *  in-process-gpu  
  *  use-gl desktop 


Also task manager shows this as separate child process. So, it seems this must be packed into any distribution with some configuration options.

**Comment by gzotti** [comment]
"Solution" for Ubuntu 22.04.01: 
sudo ln -s /usr/share /share

Now these missing resources are found at /usr/share/qt6/resources, and the plugin works. 
It seems some package master configured Qt6 wrongly (?)

**Comment by alex-w** [comment]
Please check the fresh deb packages

**Comment by worachate001** [comment]
It works. :+1: 

BTW, a lot of this warning with different ID appear in the log.

`WARNING: in StelTexture::~StelTexture() tried to delete invalid texture with ID= 10 Current GL ERROR status is 0 ( unknown GL error type )`

**Comment by alex-w** [comment]
> It works. 👍

Good news, thanks!

**Comment by alex-w** [comment]
@gzotti OK, a missing package: `apt install libqt6webengine6-data`

**Comment by gzotti** [comment]
Have you seen this? https://doc.qt.io/qt-6/qt-conf.html

**Comment by alex-w** [comment]
> Have you seen this? https://doc.qt.io/qt-6/qt-conf.html

We using qt.conf for macOS only

**Comment by worachate001** [comment]
I'm going to test version 1.22.2-3d47d9a in my Windows 10 with no Qt6 installed. Stellarium refuses to run with these errors:

The code execution cannot proceed because Qt6OpenGL.dll was not found. Including Qt6Charts.dll, Qt6OpenGLWidgets.dll and  Qt6MultimediaWidgets.dll.

**Comment by alex-w** [comment]
@worachate001 please check artefact from here: https://ci.appveyor.com/project/alex-w/stellarium/builds/44372237/job/v7tjjc13cykqbnt1

**Comment by gzotti** [comment]
> @gzotti OK, a missing package: `apt install libqt6webengine6-data`

No that is also not enough. I need that extra symlink which is a hack, not a solution. Not sure how to properly configure qt.conf (or better, qt6.conf!)

**Comment by gzotti** [comment]
https://bugs.launchpad.net/ubuntu/+source/qt6-base/+bug/1970057

**Comment by gzotti** [comment]
Probably we have to live with it and write this solution into the wiki. Even U22.04.01 has the bug.

**Comment by gzotti** [comment]
Very interesting: 

apt install qt6-base
Package qt6-base not found.

Where else would those be configured? qt6-base may just be a meta-package calling others, but is currently not existing.

**Comment by t4saha** [comment]
`export QT_XKB_CONFIG_ROOT=$RUNTIME/usr/share/X11/xkb` fixes the xkbcommon error.

**Comment by alex-w** [comment]
> https://bugs.launchpad.net/ubuntu/+source/qt6-base/+bug/1970057

https://bugreports.qt.io/browse/QTBUG-82589

**Comment by alex-w** [comment]
Hmm… the compilation on macOS via CLI having problems too when webengine is enabled :(

**Comment by gzotti** [comment]
Compilation fails? Then some Mac-specific module may be missing. If only running fails, see qt6.conf above.

**Comment by alex-w** [comment]
> Compilation fails? Then some Mac-specific module may be missing. If only running fails, see qt6.conf above.

Yes, compilation fails via CLI, but not within QtCreator :-/

**Comment by ilkant** [comment]
Stellarium-1.1-c47a4d8-x86_64.AppImage
Kubuntu 22.10
Qt 5.15.6

When I pushed bottom of screen button show online query window, Stellarium window disappeared. No error messages in log.txt file and screen. 

I write this comment because I found this thread and wanted not write a new thread about this.

**Comment by gzotti** [comment]
Despite mentioning Qt5.15, this is likely related to the discussion above.

**Comment by ilkant** [comment]
I tested compiled Stellarium (on my Kubuntu 22.10).

This is Stellarium 1.1 (v1.22.4-94821fc [master])
Compiled using GCC 12.2.0
Qt runtime version: 6.3.1
Qt compilation version: 6.3.1
Addressing mode: 64-bit

The online query window appeared to Stellarium screen and program didn't crash or disappear. And the online window worked with Wikipedia ane AAVSO.

Constellations (lines and names) did not work with this translated Stellarium. But that's another matter (bug or feature).

**Comment by alex-w** [comment]
@ilkant could you test binary package from our PPA: https://launchpad.net/~stellarium/+archive/ubuntu/stellarium-beta ?

**Comment by ilkant** [comment]
> @ilkant could you test binary package from our PPA: https://launchpad.net/~stellarium/+archive/ubuntu/stellarium-beta ?

I downloaded stellarium-qt6_23.0.0-upstream2.0-202303121537_ubuntu22.10.1_amd64.deb and stellarium-qt6-data_23.0.0-upstream2.0-202303121537_ubuntu22.10.1_all.deb. I installed them with dpkg -i on Kubuntu 22.10. Stellarium started ok and online query window worked ok. Here is log.txt
[Stellarium_log.txt](https://github.com/Stellarium/stellarium/files/10983668/Stellarium_log.txt)

**Comment by ilkant** [comment]
I downloaded appimage of  version 23.1RC2 / 23.0-1be40f4 (2023-03-19) Stellarium-23.0-1be40f4-qt5-x86_64.AppImage](https://github.com/Stellarium/stellarium-data/releases/download/weekly-snapshot/Stellarium-23.0-1be40f4-qt5-x86_64.AppImage)

I exported observation list. And then I removed one observation from list. That item was in the middle part of the list. And then I saved the list in the editor of the list.

Then I imported previously exported list from disk with import. The removed item (Plejades) didn't appear in the list.
